parachain/pallets/dot-app/src/lib.rs

```diff
@@ -77,17 +83,32 @@ decl_module! {
 
 		fn deposit_event() = default;
 
+		fn integrity_test() {
+			sp_io::TestExternalities::new_empty().execute_with(|| {
+				let allowed_decimals: &[u32] = &[10, 12];
+				let decimals = T::Decimals::get();
+				assert!(
+					allowed_decimals.contains(&decimals)
+				)
+			});
+		}
+
 		#[weight = 0]
 		#[transactional]
 		pub fn lock(origin, channel_id: ChannelId, recipient: H160, amount: BalanceOf<T>) -> DispatchResult {
 			let who = ensure_signed(origin)?;
 
 			T::Currency::transfer(&who, &Self::account_id(), amount, AllowDeath)?;
 
+			let amount_wrapped = match wrap::<T>(amount, T::Decimals::get()) {
+				Some(value) => value,
+				None => panic!("Runtime is misconfigured"),
```

Comment:
We should not panic here – quoting from the Substrate docs: 

> Runtime code should explicitly and gracefully handle all error cases, which is to say that runtime code must be "non-throwing", or must never "panic" to use Rust terminology. 

Can we instead return a runtime error, or panic during node initialization?

Reply:
Hey, yeah you're right - Will add a fix.

Reply:
Fixed in https://github.com/Snowfork/polkadot-ethereum/pull/330